common/src/test/java/io/netty/util/HashedWheelTimerTest.java

```diff
@@ -154,4 +146,72 @@ public void run(final Timeout timeout) throws Exception {
 
         timer.stop();
     }
+
+    @Test
+    public void testPendingTimeoutCountBeforeAndAfterTimeoutIsScheduledAndExecuted() throws InterruptedException {
+        HashedWheelTimer timer = new HashedWheelTimer(100, TimeUnit.MILLISECONDS);
+        timer.newTimeout(createNoOpTimerTask(), 1, TimeUnit.SECONDS);
+        long pendingTimeoutsCount = timer.pendingTimeoutsCount();
+        assertTrue("Pending timeouts count not 1 after adding 1 timeout. Actual count = "
+            + pendingTimeoutsCount, pendingTimeoutsCount == 1);
+
+        Thread.sleep(200); // Wait for next tick. This should remove timeoutToCancel1
+                          // and assign a bucket to remaining timeouts.
+        pendingTimeoutsCount = timer.pendingTimeoutsCount();
+        assertTrue("Pending timeout count is not 1 bucket is assigned to the timeout. Actual count = "
+                + pendingTimeoutsCount, pendingTimeoutsCount == 1);
+
+        Thread.sleep(3500); // Wait for timeout to execute.
+        pendingTimeoutsCount = timer.pendingTimeoutsCount();
+        assertTrue("Pending timeout count is not 0 after timeout is processed. Actual count = "
+                + pendingTimeoutsCount, pendingTimeoutsCount == 0);
+
+        timer.stop();
+    }
+
+    @Test
+    public void testPendingTimeoutCountBeforeAndAfterCancelledTimeoutIsScheduled() throws InterruptedException {
+        HashedWheelTimer timer = new HashedWheelTimer(100, TimeUnit.MILLISECONDS);
+        Timeout timeoutToCancel1 = timer.newTimeout(createNoOpTimerTask(), 1, TimeUnit.SECONDS);
+        Timeout timeoutToCancel2 = timer.newTimeout(createNoOpTimerTask(), 1, TimeUnit.SECONDS);
+        long pendingTimeoutsCount = timer.pendingTimeoutsCount();
+        assertTrue("Pending timeouts count not 2 after adding 2 timeouts. Actual count = "
+            + pendingTimeoutsCount, pendingTimeoutsCount == 2);
+
+        timeoutToCancel1.cancel();
+        Thread.sleep(200); // Wait for next tick. This should remove timeoutToCancel1
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2925)